### PR TITLE
[java] Update FieldDeclarationsShouldBeAtStartOfClass

### PR DESCRIPTION
**Pull request by oowekyala**
## Describe the PR

Update the rule FieldDeclarationsShouldBeAtStartOfClass as is.

## Related issues

- #2701 

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by travis)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to pmd/7.0.x:<br />
This changeset changes 0 violations,<br />
introduces 678 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/620dc14624b144038b9832ca5dc2bd72/diff1/index.html">Full report</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 9231 violations,<br />
introduces 8361 new violations, 1 new errors and 0 new configuration errors,<br />
removes 15871 violations, 10 errors and 2 configuration errors.<br />
<a href="https://chunk.io/pmd/620dc14624b144038b9832ca5dc2bd72/diff2/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/FieldDeclarationsShouldBeAtStartOfClass.xml [hunk 1]
```diff
@@ -64,10 +65,11 @@ public class MyTest extends AbstractJUnit4SpringContextTests {
 
     <test-code>
         <description>#1126 False positive with FieldDeclarationsShouldBeAtStartOfClass for static enums</description>
-        <expected-problems>0</expected-problems>
+        <expected-problems>3</expected-problems>
```
Comment (adangel) [inline pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/FieldDeclarationsShouldBeAtStartOfClass.xml:68]: that looks suspicious... did the default value of "ignoreEnumDeclarations" change? Hm... no, but this property doesn't seem to work anymore like before....

Comment (oowekyala) [reply pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/FieldDeclarationsShouldBeAtStartOfClass.xml]: The enum is followed by a semicolon, and the EmptyDeclaration node was not whitelisted

**Review by adangel — APPROVED** [review]
Looks good, I'll merge that now.

Thanks for investigating and fixing the build btw (https://github.com/pmd/pmd/commit/0ea8dc49f9f45c478e12dd8df326c9bcddd3c1da)

**Pull request merged**